identity/state.go

```diff
@@ -9,120 +9,131 @@ import (
 
-type IdentityState int
+type State int
 
 const (
-	IdentityStateNotSet IdentityState = iota
+	StateNotSet State = iota
 
-	IdentityStateWaitForATXSynced
-	IdentityStateRetrying
+	StateWaitForATXSynced
+	StateRetrying
 
 	// poet.
-	IdentityStateWaitingForPoetRegistrationWindow
+	StateWaitingForPoetRegistrationWindow
 	// building nipost challenge.
-	IdentityStatePoetChallengeReady
-	IdentityStatePoetRegistered
+	StatePoetChallengeReady
+	StatePoetRegistered
 	// 2w pass ...
-	IdentityStateWaitForPoetRoundEnd
-	IdentityStatePoetProofReceived
+	StateWaitForPoetRoundEnd
+	StatePoetProofReceived
 
 	// post.
-	IdentityStateGeneratingPostProof
-	IdentityStatePostProofReady
+	StateGeneratingPostProof
+	StatePostProofReady
 
 	// atx.
-	IdentityStateATXReady
-	IdentityStateATXBroadcasted
+	StateATXReady
+	StateATXBroadcasted
 
-	IdentityStateProposalPublished
+	StateProposalPublished
 )
 
-func (s IdentityState) String() string {
+func (s State) String() string {
 	switch s {
-	case IdentityStateNotSet:
+	case StateNotSet:
 		return "not set"
-	case IdentityStateWaitForATXSynced:
+	case StateWaitForATXSynced:
 		return "wait for atx synced"
-	case IdentityStateRetrying:
+	case StateRetrying:
 		return "retrying"
-	case IdentityStatePoetChallengeReady:
+	case StatePoetChallengeReady:
 		return "poet challenge ready"
-	case IdentityStateWaitingForPoetRegistrationWindow:
+	case StateWaitingForPoetRegistrationWindow:
 		return "waiting for poet registration window"
-	case IdentityStatePoetRegistered:
+	case StatePoetRegistered:
 		return "poet registered"
-	case IdentityStateWaitForPoetRoundEnd:
+	case StateWaitForPoetRoundEnd:
 		return "wait for poet round end"
-	case IdentityStatePoetProofReceived:
+	case StatePoetProofReceived:
 		return "poet proof received"
-	case IdentityStateGeneratingPostProof:
+	case StateGeneratingPostProof:
 		return "generating post proof"
-	case IdentityStatePostProofReady:
+	case StatePostProofReady:
 		return "post proof ready"
-	case IdentityStateATXReady:
+	case StateATXReady:
 		return "atx ready"
-	case IdentityStateATXBroadcasted:
+	case StateATXBroadcasted:
 		return "atx broadcasted"
-	case IdentityStateProposalPublished:
+	case StateProposalPublished:
 		return "proposal published"
 	default:
 		panic(fmt.Sprintf(ErrIdentityStateUnknown.Error()+" %d", s))
 	}
 }
 
-type IdentityStateInfo struct {
-	State        IdentityState
-	PublishEpoch *types.EpochID
-	Message      string
-	Time         time.Time
-}
+type (
+	StateInfoMetadata func(*StateInfo)
+	StateInfo         struct {
+		State        State
+		PublishEpoch *types.EpochID
+		Time         time.Time
+
+		RetryingState            *RetryingStateMetadata
+		PoetRegisteredState      *PoetRegisteredStateMetadata
+		WaitForPoetRoundEndState *WaitForPoetRoundEndStateMetadata
+		PoetProofReceivedState   *PoetProofReceivedStateMetadata
+		AtxBroadcastedState      *AtxBroadcastedStateMetadata
+		ProposalPublishedState   *ProposalPublishedStateMetadata
+	}
+)
```

Comment:
I think mixing type definitions is a bit confusing. How about:

```suggestion
type StateInfoMetadata func(*StateInfo)

type StateInfo struct {
	State        State
	PublishEpoch *types.EpochID
	Time         time.Time

	RetryingState            *RetryingStateMetadata
	PoetRegisteredState      *PoetRegisteredStateMetadata
	WaitForPoetRoundEndState *WaitForPoetRoundEndStateMetadata
	PoetProofReceivedState   *PoetProofReceivedStateMetadata
	AtxBroadcastedState      *AtxBroadcastedStateMetadata
	ProposalPublishedState   *ProposalPublishedStateMetadata
}
```

Reply:
Thanks to the linter for this change! :)

Reply:
But this makes no sense - below there is `type StateStorage struct`. Why is this not part of this block? 😅 And `make lint` doesn't complain either way, doesn't matter if its a block or not.